packages/esm-patient-forms-app/src/form-entry-interop.ts

```diff
@@ -43,3 +45,35 @@ export function launchFormEntry(
     formInfo: { encounterUuid, formUuid, visit: currentVisit },
   });
 }
+
+/**
+ * Given a list of forms and a list of HtmlFormEntryForm objects from configuration, return a List of HtmlFormEntryForm
+ * returned forms either
+ *  a) have a form resource with a name of `formEngine` and a value of `htmlformentry, or
+ *  b) have an entry in the HtmlFormEntryForm array for a given form uuid
+ * The HtmlFormEntryForm configuration provides a means to override the name and rendering mode of a given form
+ * @param allForms
+ * @param htmlFormEntryForms
+ */
+export function mapFormsToHtmlFormEntryForms(allForms: Array<Form>, htmlFormEntryForms: HtmlFormEntryForm[]) {
+  return allForms
+    ?.filter((form) => {
+      return (
+        htmlFormEntryForms?.some((hfeForm) => hfeForm.formUuid === form.uuid) ||
+        form.resources?.some((resource) => {
+          return resource.name === formEngineResourceName && resource.valueReference === htmlformentryFormEngine;
+        })
+      );
+    })
+    ?.map((form) => {
+      const hfeForm = htmlFormEntryForms?.find((f) => f.formUuid === form.uuid);
+      const simple = form.resources?.some((r) => r.name === uiStyleResourceName && r.valueReference === uiStyleSimple);
+      return {
+        formUuid: form.uuid,
+        formName: hfeForm?.formName ?? form.display ?? form.name,
+        formUiResource: hfeForm?.formUiResource,
+        formUiPage: hfeForm?.formUiPage ?? simple ? 'enterHtmlFormWithSimpleUi' : 'enterHtmlFormWithStandardUi',
+        formEditUiPage: hfeForm?.formEditUiPage ?? simple ? 'editHtmlFormWithSimpleUi' : 'editHtmlFormWithStandardUi',
+      } as HtmlFormEntryForm;
+    });
```

Comment:
Anything configured in the client-side configuration is used if specified.  Otherwise, the form attributes and resources are consulted.  Else, sensible defaults are used.